docs/maintenance/internals/systemfield.md

```diff
@@ -1,47 +1,701 @@
+        record_data = hit.to_dict()
+        creator_data = record_data.get('metadata', {}).get('creator', {})
+        version_stamp = creator_data.get('@v', '')
+
+        if version_stamp:
+            creator_id, revision_id = version_stamp.split('::')
+            current_creator = User.get_record(creator_id)
+
+            if str(current_creator.revision_id) != revision_id:
+                # Record has stale data, reindex it
+                record = MyRecord.get_record(hit.meta.id)
+                record.relations.creator.dereference()
+                record.commit()
+```
+
+## JSON vs Python: Data Format Considerations
+
+### JSON Storage Format
+
+SystemFields ultimately store data in the record's JSON dictionary. Consider how your field's data will be serialized:
+
+```python
+class DateField(SystemField):
+    """Field that handles date objects."""
+
+    def __get__(self, record, owner=None):
+        if record is None:
+            return self
+
+        date_str = self.get_dictkey(record)
+        if date_str:
+            from datetime import datetime
+            return datetime.fromisoformat(date_str)
+        return None
+
+    def __set__(self, record, value):
+        if value is not None:
+            from datetime import datetime
+            if isinstance(value, datetime):
+                # Store as ISO string in JSON
+                self.set_dictkey(record, value.isoformat(), create_if_missing=True)
+            elif isinstance(value, str):
+                # Validate string format
+                datetime.fromisoformat(value)  # Raises if invalid
+                self.set_dictkey(record, value, create_if_missing=True)
+            else:
+                raise ValueError("Date must be datetime object or ISO string")
+```
+
+### Handling Complex Objects
+
+For complex objects that don't serialize naturally to JSON:
+
+```python
+class GeolocationField(SystemField):
+    """Field for geographic coordinates."""
+
+    class GeoPoint:
+        def __init__(self, lat, lon):
+            self.lat = lat
+            self.lon = lon
+
+        def to_dict(self):
+            return {'lat': self.lat, 'lon': self.lon}
+
+        @classmethod
+        def from_dict(cls, data):
+            return cls(data['lat'], data['lon'])
+
+    def __get__(self, record, owner=None):
+        if record is None:
+            return self
+
+        data = self.get_dictkey(record)
+        if data:
+            return self.GeoPoint.from_dict(data)
+        return None
+
+    def __set__(self, record, value):
+        if value is not None:
+            if isinstance(value, self.GeoPoint):
+                self.set_dictkey(record, value.to_dict(), create_if_missing=True)
+            else:
+                raise ValueError("Value must be GeoPoint instance")
+```
+
+## Idempotence and Data Consistency
+
+SystemFields are designed to be idempotent - running the same operation multiple times should produce the same result:
```

Comment:
They are meant to be, but there is no enforcement - it's up to the developer. So I guess it's more a matter of advising developers to respect that principle :grin: : . Not sure if any example would help for that. Perhaps a note block instead? And perhaps move it to the "Best practices" section?